packages/@lwc/engine-server/src/renderer.ts

```diff
@@ -313,40 +329,42 @@ export const renderer: Renderer<HostNode, HostElement> = {
                 },
             }
         ) as CSSStyleDeclaration;
-    },
+    }
 
-    isConnected(node: HostNode) {
+    isConnected(node: HostNode): boolean {
         return !isNull(node.parent);
-    },
+    }
 
-    insertGlobalStylesheet() {
-        // Noop on SSR (for now). This need to be reevaluated whenever we will implement support for
-        // synthetic shadow.
-    },
+    insertGlobalStylesheet(content: string): void {
+        if (this.globalStylesheets.has(content)) {
+            return;
+        }
```

Comment:
Checking if the `Set` has the value is already done internally in `Set.prototype.add` no need to check that here. 

```suggestion
```